### PR TITLE
Get room properties for the signaling server from separate function.

### PR DESCRIPTION
**Pull request by fancycode**
This makes it easier to extend the properties in the future (@danxuliu might need this for #2146).

Also dispatch an event so other apps can extend the properties to return.

### CODE REVIEW

**Comment by nickvergessen** [comment]
But this is not related to the idea we had at the conference, that external signaling should not be used for one-to-one conversations, right?

**Comment by fancycode** [comment]
This is unrelated to the p2p for one-to-one conversations. #2146 is about no event being sent to guest users if the lobby flag is changed. This was because the properties didn't change and thus the signaling server didn't send out an event.

@danxuliu and I talked about adding a new property in the signaling properties for the lobby flag which will then change and trigger the event to all users. With this change, there is a central location where the room properties for the signaling server are defined.

File: lib/Room.php [hunk 1]
```diff
@@ -233,6 +233,26 @@ public function setParticipant(?string $userId, Participant $participant): void
...
+	 * @param string $userId
+	 * @return array
+	 */
+	public function getPropertiesForSignaling(string $userId): array {
+		$properties = [
+			'name' => $this->getDisplayName($userId),
+			'type' => $this->getType(),
+		];
+
+		$event = new GenericEvent($this, [
```
Comment (nickvergessen) [inline lib/Room.php]: Should use a dedicated event

File: lib/Room.php [hunk 2]
```diff
@@ -278,6 +280,23 @@ public function setParticipant(?string $userId, Participant $participant): void
...
+	/**
+	 * Return the room properties to send to the signaling server.
+	 *
+	 * @param string $userId
+	 * @return array
+	 */
+	public function getPropertiesForSignaling(string $userId): array {
+		$properties = [
+			'name' => $this->getDisplayName($userId),
+			'type' => $this->getType(),
```
Comment (nickvergessen) [inline lib/Room.php]: So I guess we need to add the lobby things here too?
@danxuliu

**Comment by nickvergessen** [comment]
So, if we add more properties and they change, it will be enough to notify guests/people when the lobby is opened, etc?

**Comment by fancycode** [comment]
> So, if we add more properties and they change, it will be enough to notify guests/people when the lobby is opened, etc?

Yes, when the properties change, an event is sent to the subscribers from the external signaling server. The signaling server doesn't inspect / evaluate the properties but simply forwards them on change.

**Comment by nickvergessen** [comment]
Oki doki, then I will add that and it should work

**Comment by nickvergessen** [comment]
> There is a system message that says that the moderator left the call.

Fixed

**Comment by nickvergessen** [comment]
So #2146 is actually fixed with this PR, the problem was the missing listener in App.vue:
```js
EventBus.$on('shouldRefreshConversations', this.refreshCurrentConversation)
```

However, since no room property changes when your participant type changes that of course does not trigger a refresh.
But I will have another look if that is easily fixable

**Comment by nickvergessen** [comment]
Not too easy without a deeper knowledge on the external signaling server, so let's merge it like this for now, as at least room info aka lobby is being handled immediatelly now

**Pull request merged**